### PR TITLE
NAS-120970 / 22.12.3 / Nas-120970: Fixes for ACL rules.

### PR DESCRIPTION
**Pull request by Raz-Dva**
Go to the ACL editor. In the Access Control List, add a new group and user. Save your changes by clicking 'Save Access Control List'. After that go to the page
_localhost:4200/credentials/users
localhost:4200/credentials/groups_
and remove the user and group that have been added to the Access Control List.
Return to the ACL editor. Deleted user and group should be marked with 'Ace has errors' icon
This check is needed for both NFSv4 and POSIX types

### CODE REVIEW

**Comment by RehanY147** [comment]
I can't create new users to test this issue. The user form is bugged. Blocked. https://ixsystems.atlassian.net/browse/NAS-121606

**Comment by undsoft** [comment]
@RehanY147 PR to fix user form just got merged.

File: src/app/pages/datasets/modules/permissions/stores/dataset-acl-editor.store.ts [hunk 1]
```diff
@@ -66,6 +66,15 @@ export class DatasetAclEditorStore extends ComponentStore<DatasetAclEditorState>
           this.ws.call('filesystem.stat', [mountpoint]),
         ]).pipe(
           tap(([acl, stat]) => {
+            acl.acl.forEach((ace, aceIndex) => {
+              if ((!ace.who && [NfsAclTag.User, PosixAclTag.User].includes(ace.tag))
```
Comment (RehanY147) [inline src/app/pages/datasets/modules/permissions/stores/dataset-acl-editor.store.ts:70]: Let's make these conditionals into named variables that explain what this condition means. such as 
`const aceIsUser = !ace.who && [NfsAclTag.User, PosixAclTag.User].includes(ace.tag);
...
if(aceIsUser && ...)`

**Review by RehanY147 — CHANGES_REQUESTED** [review]

File: src/app/pages/datasets/modules/permissions/stores/dataset-acl-editor.store.ts (hunk 1 above)
Comment (Raz-Dva) [reply src/app/pages/datasets/modules/permissions/stores/dataset-acl-editor.store.ts]: I have added two constants` unknownUser`, `unknownGroup`. When the store is loaded, a check is made to see if the deleted user or group exists (`!ace.who`).

**Review by RehanY147 — APPROVED** [review]

**Pull request merged**